plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/AggregationQueryPageSource.java

```diff
@@ -0,0 +1,260 @@
+        boolean hasMetricAggregation = false;
+        for (Aggregation agg : aggregations) {
+            if (agg instanceof CompositeAggregation) {
+                hasBucketsAggregation = true;
+                if (hasMetricAggregation) {
+                    throw new IllegalStateException("Bucket and metric aggregations should not be both present.");
+                }
+                CompositeAggregation compositeAggregation = (CompositeAggregation) agg;
+                for (CompositeAggregation.Bucket bucket : compositeAggregation.getBuckets()) {
+                    Map<String, Object> line = new HashMap<>();
+                    line.putAll(bucket.getKey());
+                    for (Aggregation metricAgg : bucket.getAggregations()) {
+                        if (metricAgg instanceof NumericMetricsAggregation.SingleValue) {
+                            NumericMetricsAggregation.SingleValue sv = (NumericMetricsAggregation.SingleValue) metricAgg;
+                            line.put(metricAgg.getName(), extractSingleValue(sv));
+                        }
+                        else if (metricAgg instanceof Stats) {
+                            line.put(metricAgg.getName(), extractSumFromStatsValue((Stats) metricAgg));
+                        }
+                    }
+                    result.add(line);
+                }
+            }
+            else if (agg instanceof NumericMetricsAggregation.SingleValue) {
+                hasMetricAggregation = true;
+                if (hasBucketsAggregation) {
+                    throw new IllegalStateException("Bucket and metric aggregations should not be both present.");
+                }
+                NumericMetricsAggregation.SingleValue sv = (NumericMetricsAggregation.SingleValue) agg;
+                singleValueMap.put(agg.getName(), extractSingleValue(sv));
+            }
+            // We use stats aggregation instead of sum aggregation
+            else if (agg instanceof Stats) {
+                hasMetricAggregation = true;
+                if (hasBucketsAggregation) {
+                    throw new IllegalStateException("Bucket and metric aggregations should not be both present.");
+                }
+                singleValueMap.put(agg.getName(), extractSumFromStatsValue((Stats) agg));
+            }
+            else {
+                throw new IllegalStateException("Unrecognized aggregation type.");
+            }
+        }
+        if (hasBucketsAggregation) {
+            return result.build();
+        }
+        else {
+            return ImmutableList.of(singleValueMap);
+        }
+    }
+
+    private Double extractSumFromStatsValue(Stats statsValue)
+    {
+        double sumValue = statsValue.getSum();
+        // Before 7.0.0, sum(all null) in the result of stats aggregation is null, and the following commit fix it
+        // https://github.com/elastic/elasticsearch/commit/55b9dfdd52ea96d3f51f719cb65606ca18a97277
```

Comment:
I'm a bit confused by the comment.  Are you saying that before version 7.0.0, the sum aggregation had the right semantics (returned null)?  And the commit above changed that?  If that's the case, we shouldn't say the commit "fixed" it (at least from Trino's perspective 😉). Perhaps I misunderstood?

Reply:
@aalbu Yes. The commit belongs to server code. But on the client side, the result of sum in stats aggregation always parse null to 0. 
So i decide to remove this case finally, which means we need always check the result of `min` regardless elasticsearch server version. 😉